sleap_nn/data/providers.py

```diff
@@ -0,0 +1,29 @@
+"""Handle importing of sleap data."""
+import torchdata.datapipes.iter as dp
+import lightning.pytorch as pl
+
+import sleap_io as sio
+
+
+class LabelsReader(dp.IterDataPipe):
+    """Datapipe for reading frames from Labels object."""
+
+    def __init__(self, labels: sio.Labels):
+        """Initialize LabelsReader.
+
+        Args:
+            labels: sleap_io.Labels object that contains LabeledFrames that will be
+                accessed through a torchdata DataPipe
+        """
```

Comment:
Consider commenting instance attributes as part of class-level docstring as per Google style docstrings ([ref](https://stackoverflow.com/questions/37019744/is-there-a-consensus-on-what-should-be-documented-in-the-class-and-init-docs)):

> Classes should have a docstring below the class definition describing the class. If your class has public attributes, they should be documented here in an Attributes section and follow the same formatting as a function’s Args section.